graphql/src/main/java/com/scalar/db/graphql/datafetcher/QueryScanDataFetcher.java

```diff
@@ -14,38 +14,61 @@
 import com.scalar.db.api.Scan.Ordering.Order;
 import com.scalar.db.exception.storage.ExecutionException;
 import com.scalar.db.exception.transaction.TransactionException;
-import com.scalar.db.graphql.schema.TableGraphQlModel;
 import com.scalar.db.io.Key;
+import graphql.schema.DataFetcher;
 import graphql.schema.DataFetchingEnvironment;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 
-public class QueryScanDataFetcher extends DataFetcherBase<Map<String, List<Map<String, Object>>>> {
+public class QueryScanDataFetcher implements DataFetcher<Map<String, List<Map<String, Object>>>> {
+  private final DistributedStorage storage;
+  private final DataFetcherHelper helper;
 
-  public QueryScanDataFetcher(DistributedStorage storage, TableGraphQlModel tableModel) {
-    super(storage, tableModel);
+  public QueryScanDataFetcher(DistributedStorage storage, DataFetcherHelper helper) {
+    this.storage = storage;
+    this.helper = helper;
   }
 
-  @SuppressWarnings("unchecked")
   @Override
   public Map<String, List<Map<String, Object>>> get(DataFetchingEnvironment environment)
       throws Exception {
     Map<String, Object> scanInput = environment.getArgument("scan");
+    Scan scan = createScan(scanInput);
+
+    // TODO: scan.withProjections()
```

Comment:
Is this forgotten? or just a TODO for upcoming PRs?

Reply:
Thank you! It’s going to be implemented in another PR.